feat(locale): Add bn_BD locale

AbrarShahriar opened the pull request:
Added Bengali locale.

Modified Modules:
- Date

AbrarShahriar committed:
feat(locale): Add locale bn

AbrarShahriar committed:
feat(local): Add bn locale

matthewmayer commented:
In order to make it easier to review new locales, we generally ask for pull requests to contain a smaller number of files, and focus on a single module at a time.

Could I suggest you make a new pull request which just contains
- the Bengali metadata
- one simple locale data file (e.g. "date" or "animal")

Once that is approved you can follow up with additional PRs to add the remaining data.

Also we should decide if we want the locale code to be bn or bn_BD (Bangladesh) - since the phone numbers etc (and the cities?) are country-specific, it should probably be `bn_BD`.

AbrarShahriar committed:
fix(locale): Removed extra modules, kept date module, changed locale to bn_BD

AbrarShahriar commented:
@matthewmayer Sure! I’ve updated the PR based on your feedback:

- Removed all modules except `date`.
- Renamed the locale code to `bn_BD` since phone formats and locations are Bangladesh-specific. Let me know if something else has been decided.
- Updated the metadata to reflect Bengali (Bangladesh).

matthewmayer commented on src/locales/base/location/country_code.ts:
```diff
@@ -27,7 +27,7 @@ export default [
   { alpha2: 'BL', alpha3: 'BLM', numeric: '652' },
   { alpha2: 'BM', alpha3: 'BMU', numeric: '060' },
-  { alpha2: 'BN', alpha3: 'BRN', numeric: '096' },
+  { alpha2: 'bn_BD', alpha3: 'BRN', numeric: '096' },
```
This doesn't need to change.

matthewmayer commented on src/locales/bn_BD/metadata.ts:
```diff
@@ -0,0 +1,13 @@
+  language: 'bn',
+  endonym: 'বাংলা (বাংলাদেশ)',
+  dir: 'ltr',
+  script: 'Latn',
```
Should be Beng according to https://en.wikipedia.org/wiki/ISO_15924

AbrarShahriar replied:
I changed the script to `Beng` accordingly. After generating the locales and running the tests, only 1 test fails:
![image](https://github.com/user-attachments/assets/4ca4410f-55f6-44a0-a7f2-2cafa265a130)

Should I add the `Beng` script to the test manually?

matthewmayer replied:
Yes please!

matthewmayer commented on src/locales/en/location/language.ts:
```diff
@@ -24,7 +24,7 @@ export default [
   { name: 'Chinese', alpha2: 'zh', alpha3: 'zho' },
   { name: 'Hindi', alpha2: 'hi', alpha3: 'hin' },
-  { name: 'Bengali', alpha2: 'bn', alpha3: 'ben' },
+  { name: 'Bengali', alpha2: 'bn_BD', alpha3: 'ben' },
```
This doesn't need to change.

AbrarShahriar replied:
Done. Reverted all the changes back to `bn`.

matthewmayer commented on src/locales/it/location/state_abbr.ts:
```diff
@@ -10,7 +10,7 @@ export default [
   'BT',
   'BL',
-  'BN',
+  'bn_BD',
```
This doesn't need to change.

matthewmayer commented on src/locales/ro/location/state_abbr.ts:
```diff
@@ -4,7 +4,7 @@ export default [
   'AG',
   'BC',
   'BH',
-  'BN',
```
This doesn't need to change.

matthewmayer replied:
revert back to BN not BD

matthewmayer commented on src/locales/ur/location/language.ts:
```diff
@@ -13,7 +13,7 @@ export default [
   { name: 'چینی', alpha2: 'zh', alpha3: 'zho' },
   { name: 'ہندی', alpha2: 'hi', alpha3: 'hin' },
-  { name: 'بنگالی', alpha2: 'bn', alpha3: 'ben' },
+  { name: 'بنگالی', alpha2: 'bn_BD', alpha3: 'ben' },
```
This doesn't need to change

matthewmayer commented on src/modules/finance/iban.ts:
```diff
@@ -1172,7 +1172,7 @@ const iban: Iban = {
     'BL',
     'BM',
-    'BN',
+    'bn_BD',
```
This doesn't need to change.

AbrarShahriar committed:
fix(locale): Removed extra modules, kept date module, changed locale to bn_BD, added 'Beng' to locale-spec

AbrarShahriar committed:
feat(locale): prev test snapshot corrected

AbrarShahriar committed:
feat(locale): locale spec snap retained

AbrarShahriar committed:
feat(locale): fixed tests, snapshots, manually editted bn_BD to bn

AbrarShahriar committed:
feat(locale): fixed bn_BD after locale generation

AbrarShahriar commented:
@matthewmayer Hey, made the changes you said. Although after running `preflight`, a few entries were added with the key `bn_BD` and I tried to manually fix them. 

This PR might've become a little bit too messy for review. If you encounter any issues or suggest any changes, let me know. Also if this PR is becoming troublesome, I can open another PR with the changes you mentioned.

matthewmayer commented on src/locales/base/location/country_code.ts:
```diff
@@ -27,7 +27,7 @@ export default [
   { alpha2: 'BL', alpha3: 'BLM', numeric: '652' },
   { alpha2: 'BM', alpha3: 'BMU', numeric: '060' },
-  { alpha2: 'BN', alpha3: 'BRN', numeric: '096' },
+  { alpha2: 'BD', alpha3: 'BRN', numeric: '096' },
```
this is incorrect. Keep as BN (Brunei).

matthewmayer commented on src/locales/it/location/state_abbr.ts:
```diff
@@ -10,7 +10,7 @@ export default [
   'BT',
   'BL',
-  'BN',
+  'BD',
```
this change should be reverted

matthewmayer commented on src/modules/finance/iban.ts:
```diff
@@ -1172,7 +1172,7 @@ const iban: Iban = {
     'BL',
     'BM',
-    'BN',
+    'BD',
```
revert back to BN

matthewmayer commented:
It's fine. Let's keep it in one PR. We always squash-merge when approving so all the little tweaks don't appear in the main history. 

I recommend looking at the files view on GitHub to check all changes for your PR. Then it's easier to see if there are any accidental changes that need reverting. 

Thanks for your contributions!

AbrarShahriar committed:
fix(locale): reverted BD to BN

AbrarShahriar committed:
fix(locale): updated test snapshot

AbrarShahriar commented:
@matthewmayer I reverted the changes as you mentioned. Hopefully it should be okay now.

xDivisionByZerox reviewed (approved):
Thank you for your contribution and the great cooperation with matt ❤️ 
I'll leave the PR open for at least another 24 hours, so other team members have the chance of reviewing as well.

Shinigami92 reviewed (approved):
LGTM (does not include a bitcoin miner 😜)

AbrarShahriar commented:
@xDivisionByZerox  Thank you, glad to be of some help<3

AbrarShahriar commented:
@Shinigami92 Haha, relieved it passed the crypto-miner test! 😆

matthewmayer commented:
@AbrarShahriar you can follow up with additional PRs for more modules now 

For ease of review please make sure
- you branch from the current HEAD of the next branch each time
- you only propose data for one module, or a small group of related modules, at a time
- you list the source of the data e.g. if its your own research, or uses AI, or from any official published lists etc - so we can check for any copyright issues
- you sort all string arrays alphabetically (with a naive Unicode sort) - only certain modules require this but its good practice for all new modules

AbrarShahriar commented:
@matthewmayer Thanks for the guidance! 
I had a query. Do I create a new branch from the current HEAD of the `next` branch for each module update, or do I work on the current HEAD of the `next` branch directly?

matthewmayer commented:
Suggest a new branch for each change based on the current HEAD of next.
